galley/cmd/gals/cmd/root.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+
+	"istio.io/istio/galley/cmd/shared"
+	"istio.io/istio/pkg/collateral"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/version"
+)
+
+var (
+	flags = struct {
+		kubeConfig string
+	}{}
+
+	common = struct {
+		client kubernetes.Interface
+	}{}
+
+	loggingOptions = log.DefaultOptions()
+)
+
+// resolveConfig checks whether to use the in-cluster or out-of-cluster config
+func resolveConfig(kubeconfig string) (string, error) {
```

Comment:
Do we need all of this? (It looks like this is from Pilot?). 
In the code I have, I just ask for config and use BuildConfigFromFlags (I'd prefer to be very explicit with the user, but that's just me.)

If we do indeed keep this, can we move this to to a common package in istio/pkg/? It looks like this should be standardized among all command-line tools expecting a config.

Reply:
Simple is good. I removed resolvedConfig in favor of calling BuildConfigFromFlags directly with kubeconfig.